helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -1309,22 +1306,24 @@ public void run() throws Exception {
     }
   }
 
-  private void fireChildChangedEvents(final String path, Set<IZkChildListener> childListeners) {
+  private void fireChildChangedEvents(final String path, Set<IZkChildListener> childListeners, boolean pathExists) {
     try {
-      final ZkPathStatRecord pathStatRecord = new ZkPathStatRecord(path);
       for (final IZkChildListener listener : childListeners) {
         _eventThread.send(new ZkEvent("Children of " + path + " changed sent to " + listener) {
           @Override
           public void run() throws Exception {
-            // Reinstall watch before listener callbacks to check the znode status
-            if (!pathStatRecord.pathChecked()) {
```

Comment:
This will remove our previous improvement for multiple listeners. Why we remove it?

Reply:
You can just change the condition from hasListeners(path) to pathExists && hasListeners(path), right?

Reply:
I checked the code in `recordPathStat`, basically, the line `pathStatRecord.recordPathStat(getStat(path, hasListeners(path)), OptionalLong.empty());` does nothing! This is because notificationTime is optional and the propogation latency is never recorded whne child change events. 

I feel like leaving this piece of code here will just make ourselves confused. And if in the future, we do need to record propagation latency for child events. We could add it back. 

Second, `getStat(path, hasListeners(path))` will add a exists watcher for the parent path (child listener). Don't we want to add watcher type of `child`? The code here looks problematic for me, please clarify the original purpose if I misunderstand it. Thanks!

Reply:
Seems to me that we can keep the same style as fireDataChangeEvent

`
if (!pathStatRecord.pathChecked()) {
   pathStatRecord.recordPathStat(getStat(path, pathExists), OptionalLong.empty());
}

// The rest leave it the same 
`

Reply:
@kaisun2000 Yes. However, as I replied earlier, this line `pathStatRecord.recordPathStat(getStat(path, pathExists), OptionalLong.empty());` does nothing (need to read the exact implemenation) when `OptionalLong.empty()` is passed in. We're not recording propogation latency for child listeners anyway. It only confuses us AND the `getStat()` method will possibly add another leaked watch

Reply:
I see. Is it because `if (_monitor != null && stat != null && **notificationTime.isPresent())**` in the pathStatRecord.recordPathState() implementation?

Then my question is that before this diff, line 1321 `pathStatRecord.recordPathStat(getStat(path, hasListeners(path)), OptionalLong.empty());` still does not really emit metrics either. So my proposal keeps the old behavior? Or did I miss something here?

Reply:
@kaisun2000 Yes, keep the same code is fine; Is it really necessary? Why not add it later once we do need to record propagation latency?

Reply:
This is actually debatable. I am more into thinking that " the fix" just fix the problem, don't introduce non essential change. Or change the metric reporting behavior. One fix for one issue.

That said, I don't have strong opinion here. Check with JJ. I am fine either way.

Reply:
all right, I will revert this part of change